code/__DEFINES/mobs.dm

```diff
@@ -468,3 +468,15 @@
 
 //Saves a proc call, life is suffering. If who has no targets_from var, we assume it's just who
 #define GET_TARGETS_FROM(who) (who.targets_from ? who.get_targets_from() : who)
+
+///Ensures mobs that have recently stopped crawling can be easily hit for a little longer. See below.
+#define CRAWLING_PENALTY (0.3 SECONDS + world.tick_lag*2)
+
+/**
+ * Whether a projectile or thrown thing with hit_crawling_targets TRUE
+ * should hit a mob that's currently on movement cooldown while lying down. (1300+50) <= 1300
+ */
+#define IS_HITTING_DECK(living, delay, skip_trait) (living.body_position == LYING_DOWN && (delay + CRAWLING_PENALTY) <= world.time && (skip_trait || !HAS_TRAIT(living, TRAIT_CANNOT_EVADE_PROJECTILES)))
```

Comment:
So the idea is to allow people to dodge projectiles only while moving? or while not moving?